src/api/drone.rs

```diff
@@ -0,0 +1,6 @@
+use crate::types::topology::routes::Path;
+
+pub struct Packet {}
+pub trait DroneAble {
+    fn forward_packet(&self, path: &Path, hop: usize, packet: Packet) -> bool;
```

Comment:
is the path field necessary? isn't it already comprehended in the source header part of the packet field?